_posts/2017-05-31-khalid's-initial-thoughts-on-vuejs.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title: "Khalid's Initial Thoughts On VueJs"
+date: 2017-05-31 11:43:15
+tags: [ "vuejs", "JavaScript", "frontend", "clientside" ]
+categories:
+twitter_text: "@buhakmeh initial thoughts on @vuejs"
+authors: Khalid Abuhakmeh
+image: https://vuejs.org/images/logo.png
+image_credit: VueJs
+image_url: https://vuejs.org
+---
+
+The RIMdev team composition is a unique blend of disciplines forged in the crucible of time. We have backend developers, frontend developers, designers, quality assurance engineers, and data engineers. With that cauldron of experience, it isn't a suprise that their are a multitude of opinions about the best way to build software in 2017. From my perspective, here is a short list of consensus:
+
+1. User experience über alles (UX is king).
+1. Centralizing and *"Normalizing"* data access via APIs.
+1. Thin clients with limited business logic is ideal.
+1. Speed is a feature.
+1. Static site generation where ever possible.
+
+Open questions:
+
+1. Do we want to exclusively build SPAs?
+1. Which client-side frameworks / libraries to use? (Angular, React, and/or VueJs)
+1. When should we move to ASP.NET Core? (I think we can soon)
+1. Docker?
+
+This post focuses on the client-side questions and specifically my thoughts on [VueJs](https://vuejs.org). 
+
+## Do We Want To Exclusively Build SPAs?
+
+SPAs stir up a tempest of emotions in me. The first emotion is excitement:
+
+>  "Cool, a new tool! It will solve all my problems and I love a challenge." 
+
+The second emotion is dread:
+
+> "What if everything goes wrong and this makes everything worse?".
+
+Caution being the last emotion: 
+
+> "I should think about this..." 
+
+Should we build only SPAs? The short answer is `no`, but the long answer is more complicated.
+
+For Ritter Insurance Marketing, there are two kinds of web properties our team builds: `Marketing Sites` and `Applications`. For marketing properties, it makes sense to sparingly use JavaScript in the form ofcomponents. Components give the appearance of a dynamic site while enhancing an otherwise static experience. This blog is a good example of that. Our search component written by Nathan and it works amazingly. `Applications` tend to have more interactions and center around complex resources. Applications is where it makes sense to leverage a SPA approach.
```

Comment:
Please revise `Our search component written by Nathan and it works amazingly.`.

`is written by` perhaps?